regression.py

```diff
@@ -0,0 +1,157 @@
+        Keccak(var="xkcp"),  # armv7m
+        ntt_dilithium(),  # armv7m
+        ntt_kyber(),  # armv7m
+        ntt_kyber_1_23_45_67(),  # armv8m
+        ntt_dilithium_12_34_56_78(),  # armv8m
+    )
+
+    all_regression_test_names = [e.name for e in regression_tests]
+
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+    parser.add_argument(
+        "--regression_tests",
+        type=str,
+        default="all",
+        help=f"The list of regression_tests to be run, comma-separated list from "
+        f"{all_regression_test_names}. "
+        f"Format: {{name}}_{{variant}}_{{target}}, e.g., "
+        "ntt_kyber_123_4567_scalar_load_a55",
+    )
+    parser.add_argument("--dry-run", default=False, action="store_true")
+    parser.add_argument("--debug", default=False, action="store_true")
+    parser.add_argument("--silent", default=False, action="store_true")
+    parser.add_argument("--iterations", type=int, default=1)
+    parser.add_argument("--timeout", type=int, default=0)
+    parser.add_argument("--debug-logfile", type=str, default=None)
+    parser.add_argument("--log-model", default=False, action="store_true")
+    parser.add_argument("--log-model-dir", type=str, default="models")
+    parser.add_argument(
+        "--only-target",
+        type=str,
+        choices=[
+            Target_CortexM7.__name__,
+            Target_CortexM55r1.__name__,
+            Target_CortexM85r1.__name__,
+            Target_CortexA55.__name__,
+            Target_CortexA72.__name__,
+            Target_AppleM1_firestorm.__name__,
+            Target_AppleM1_icestorm.__name__,
+            Target_AArch64Big.__name__,
+        ],
+    )
+    args = parser.parse_args()
+    if args.regression_tests != "all":
+        todo = args.regression_tests.split(",")
+    else:
+        todo = all_regression_test_names
+    iterations = args.iterations
+
+    def run_reg_test(name, **kwargs):
+        ex = None
+        for e in regression_tests:
+            if e.name == name:
+                ex = e
+                break
+        if ex is None:
+            raise OptimizationRunnerException(f"Could not find regression test {name}")
+        start_time = time.time()
+        for _ in range(iterations):
+            ex.run(**kwargs)
+        end_time = time.time()
+
+        avg_time = (end_time - start_time) / iterations
+
+        return e.name, avg_time
+
+    reg_res = {}
+
+    for e in todo:
+        n, t = run_reg_test(
+            e,
+            debug=args.debug,
+            dry_run=args.dry_run,
+            silent=args.silent,
+            log_model=args.log_model,
+            debug_logfile=args.debug_logfile,
+            log_model_dir=args.log_model_dir,
+            timeout=args.timeout,
+            only_target=args.only_target,
+        )
+        reg_res[n] = t
+
+    summary_file = os.environ.get("GITHUB_STEP_SUMMARY")
+    with open(summary_file, "w") as f:
```

Comment:
can you wrap this in an 
`if sumary_file is not None:` 
please? otherwise this errors outside of CI.